File: Implementations/splay_tree_test.cpp
#include "splay_tree.h"
#include <cstdlib>
#include <time.h>
using namespace std;

int main()
{
  splay_tree<int> arbol;

  /* FOR MANUAL TESTING!
  int  opcode = 2, aux;

  while(opcode != -1)
  {
    cin >> aux;

    try
    {
      if(opcode == 0)      arbol.print();
      else if(opcode == 1) arbol.find(aux);
      else if(opcode == 2) arbol.insert(aux);
      else if(opcode == 3) arbol.erase(aux);
      else if(opcode == 4) {if(arbol.empty())  cout << "VACIO\n";
                            else               cout << "NO VACIO\n";}
    }
    catch (const char* msg)
    {
      cerr << msg << endl;
    }
    catch (exception &e)
    {
      cout << e.what() << endl;
    }

    cin >> opcode;
  }*/

  /* FOR AUTOMATIC TEST
  srand(time(NULL));

  int upLimit = 1000, nextValue;

  try
  {
    for(int i = 0; i < 1000; i++)
    {
      nextValue = rand()%upLimit;
      arbol.insert(nextValue);
    }

    for(int i = 0; i < 1000; i++)
    {
      nextValue = rand()%upLimit;
      arbol.find(nextValue);
    }

    for(int i = 0; i < 4000; i++)
    {
      nextValue = rand()%upLimit;
      arbol.erase(nextValue);
    }
  }
  catch (const char* msg)
  {
    cerr << msg << endl;
  }
  catch (exception &e)
  {
    cout << e.what() << endl;
  }

  arbol.print(); */

  return 0;
}

File: Implementations/SplayTreeTest.java
package ucm.fdi.marp;

import ucm.fdi.marp.SplayTree;
import java.util.Random;
import java.util.Scanner;

public class SplayTreeTest
{
	public static void main(String args[])
	{

		/* 
		*       FOR AUTOMATIC TESTING
		*
		SplayTree<Integer> arbol = new SplayTree<Integer>();

		Random rand = new Random();

		int upLimit = 1000, nextValue;

		try
		{
		  for(int i = 0; i < 1000; i++)
		  {
		    nextValue = rand.nextInt(upLimit);
		    arbol.insert(nextValue);
		  }

		  for(int i = 0; i < 1000; i++)
		  {
		    nextValue = rand.nextInt(upLimit);
		    arbol.find(nextValue);
		  }

		  for(int i = 0; i < 4000; i++)
		  {
		    nextValue = rand.nextInt(upLimit);
		    arbol.erase(nextValue);
		  }
		}
		catch (Exception e)
		{
		  System.out.println(e.getMessage());
		}

		arbol.print(); */

		/*
		*   	FOR MANUAL TESTING
        *
	    int  opcode = 2, aux;
		Scanner in = new Scanner(System.in);

		while(opcode != -1)
		{
			aux = in.nextInt();

			try
			{
				if(opcode == 0)      arbol.print();
				else if(opcode == 1) arbol.find(aux);
				else if(opcode == 2) arbol.insert(aux);
				else if(opcode == 3) arbol.erase(aux);
				else if(opcode == 4) {if(arbol.empty())  System.out.println("VACIO");
									  else               System.out.println("NO VACIO");}
			}
			catch (Exception e)
			{
				System.out.println(e.getMessage());
			}

			opcode = in.nextInt();
		}*/
	}
}
